hal/api/RawSerial.h

```diff
@@ -32,6 +32,8 @@ namespace mbed {
  * Can be used for Full Duplex communication, or Simplex by specifying
  * one pin as NC (Not Connected)
  *
+ * @Note Synchronization level: Not protected
```

Comment:
Ok, so why instantiate the mutex in this class then? Also why doesn't it have the comment that some others have then: "Set by subclass"?